services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/resilience/StormLcmSpec.groovy

```diff
@@ -47,30 +64,29 @@ class StormLcmSpec extends BaseSpecification {
         assert wfmContainer
         dockerClient.restartContainer(wfmContainer.id())
         dockerClient.waitContainer(wfmContainer.id())
-        //wait until it starts to respond
-        Wrappers.wait(15) {
-            northbound.activeSwitches
-            true
-        }
+        TimeUnit.SECONDS.sleep(WFM_WARMUP_SECONDS)
 
-        then: "Network topology is unchanged"
-        northbound.activeSwitches.size() == topology.activeSwitches.size()
-        northbound.getAllLinks().findAll {
-            it.state == IslChangeType.DISCOVERED
-        }.size() == topology.islsForActiveSwitches.size() * 2
+        then: "Database nodes and relations are unchanged"
+        def newNodes = database.dumpAllNodes()
+        def newRelation = database.dumpAllRelations()
+        expect newNodes, sameBeanAs(nodesDump)
+        expect newRelation, sameBeanAs(relationsDump).ignoring("time_modify")
 
-        and: "Flow remains valid"
-        northbound.validateFlow(flow.id).each { direction ->
-            assert direction.discrepancies.findAll { it.field != "meterId" }.empty
+        and: "Flows remain valid in terms of installed rules and meters"
+        flows.each { flow ->
+            northbound.validateFlow(flow.id).each { direction ->
+                assert direction.discrepancies.findAll { it.field != "meterId" }.empty
+            }
         }
 
         and: "Flow can be updated"
-        flowHelper.updateFlow(flow.id, flow.tap { it.source.vlanId++ })
-        northbound.validateFlow(flow.id).each { direction ->
+        def flowToUpdate = flows[0]
+        flowHelper.updateFlow(flowToUpdate.id, flowToUpdate.tap { it.source.vlanId++ })
```

Comment:
It looks like such a VLAN update can lead to conflicts. Can we update flow more gracefully to avoid it?

Reply:
nice catch